templates/ad-1.template

```diff
@@ -278,20 +283,21 @@ Parameters:
     MinLength: '5'
     Type: String
   DomainDNSName:
-    AllowedPattern: '^[a-zA-Z0-9\-\.]+$'
+    AllowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
     Default: example.com
     Description: Fully qualified domain name (FQDN) of the forest root domain e.g. example.com
-    MaxLength: '255'
-    MinLength: '2'
+    MaxLength: '25'
+    MinLength: '3'
```

Comment:
Why 3?  The actual min is 2.

https://learn.microsoft.com/en-us/troubleshoot/windows-server/identity/naming-conventions-for-computer-domain-site-ou#dns-domain-names